packages/flutter_tools/lib/src/test/test_device.dart

```diff
@@ -17,7 +16,10 @@ abstract class TestDevice {
   /// Starts the test device with the provided entrypoint.
   ///
   /// Returns a channel that can be used to communicate with the test process.
-  Future<StreamChannel<String>> start({@required String compiledEntrypointPath});
+  ///
+  /// It is up to the device to determine if [entrypointPath] is a precompiled
+  /// or raw source file.
```

Comment:
Unfortunately the abstraction here isn't very ideal.

For flutter_test, we can accept the precompiled dill, or if we move the compiler into `FlutterTesterTestDevice`, we can accept Dart source files.

For a raw `Device`, it accepts a source file, or a precompiled APK/IPA.

I can't quite think of a way to abstract this better given the awkward compilation of sources that happen in the `FlutterTestPlatform` for the `flutter_tester`. let me know if there's any better ideas.

Reply:
I think its OK, provided you document the convention.

Reply:
Updated the doc comment for this